projects/laji/src/app/shared-modules/select/select-subcategories/select-subcategories.component.ts

```diff
@@ -0,0 +1,607 @@
+import {
+  ChangeDetectionStrategy,
+  ChangeDetectorRef,
+  Component,
+  ElementRef,
+  EventEmitter,
+  Input,
+  OnChanges,
+  OnDestroy,
+  OnInit,
+  Output,
+  ViewChild
+} from '@angular/core';
+import { interval as ObservableInterval, Subject, Subscription } from 'rxjs';
+import { debounceTime, take, takeUntil } from 'rxjs/operators';
+import { FilterService } from '../../../shared/service/filter.service';
+import { Router, ActivatedRoute, ActivatedRouteSnapshot } from '@angular/router';
+
+export interface SelectOptions {
+  id: string;
+  value: string;
+  category: string;
+  info?: string;
+  checkboxValue: boolean|undefined;
+}
+
+@Component({
+  selector: 'laji-select-subcategories',
+  templateUrl: './select-subcategories.component.html',
+  styleUrls: ['./select-subcategories.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class SelectSubcategoriesComponent implements OnInit, OnChanges, OnDestroy {
+  private unsubscribe$ = new Subject<null>();
+
+  @Input() options: [{[key: string]: SelectOptions[]}];
+  @Input() title: string;
+  @Input() filterPlaceHolder = 'Search...';
+  @Input() useFilter = true;
+  @Input() selected: object = {id: [], category: undefined};
+  @Input() open = false;
+  @Input() disabled = false;
+  @Input() outputOnlyId = false;
+  @Output() selectedChange = new EventEmitter<object>();
+  @Input() multiple = true;
+  @Input() info: string;
+  @Input() loading = false;
+  @Input() subCategories = [];
+  @Input() subTitleBase = '';
+  @Input() filtersName = [];
+  @Input() filtersValues = [];
+  @ViewChild('filter') filter: ElementRef;
+
+  @Output() update = new EventEmitter<{id: string[] | string, category: string}>();
+
+  selectedOptions = {};
+  unselectedOptions = {};
+  tmpSelectedOption = {};
```

Comment:
could you add types to these three. It would help to clarify initOptions method bellow